nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-service.js

```diff
@@ -865,7 +904,7 @@
                     referencingComponentRevisions[referencingComponentEntity.id] = nfClient.getRevision(referencingComponentEntity);
                 }
             } else {
-                if (referencingComponent.referenceType === 'Processor' || referencingComponent.referenceType === 'ReportingTask') {
+                if (referencingComponent.referenceType === 'Processor' || referencingComponent.referenceType === 'ReportingTask' || referencingComponent.referenceType === 'FlowAnalysisRule') {
```

Comment:
If an Enabled Flow Analysis Rule references a Controller Service and that service is Disabled, does the Flow Analysis Rule also need to be Disabled?

If not, this is unneeded.

If yes, this will attempt to include it in subsequent calls to update a Controller Service Referencing Components. This happens when a user Disables a Controller Service or when a user Enabled a Controller Service and selects Service and Referencing Components. It would also mean that the back end would need to consider Flow Analysis Rules in this API. I don't think it does currently.

Reply:
Looking a little more closely here, I think we'd need to make some minor changes to ensure the Flow Analysis Rules are not passed to the back end with the other schedulable components. Instead, it may make sense to include them with the referencing Services but we'd probably want to look at it a little more closely.

Reply:
Flow Analysis Rules are supposed to work similar to other Controller Service referencing components. If the Controller Service is disabled the Rule should be disabled as well.

Both frontend and backend can handle this with the current implementation.